--- a/arch/x86/lib/iomem.c
+++ b/arch/x86/lib/iomem.c
@@ -95,7 +95,7 @@
 
 void memcpy_fromio(void *to, const volatile void __iomem *from, size_t n)
 {
-	if (cc_platform_has(CC_ATTR_GUEST_UNROLL_STRING_IO))
+	if (IS_ENABLED(CONFIG_KOBOX_HOSTED) || cc_platform_has(CC_ATTR_GUEST_UNROLL_STRING_IO))
 		unrolled_memcpy_fromio(to, from, n);
 	else
 		string_memcpy_fromio(to, from, n);
@@ -104,7 +104,7 @@
 
 void memcpy_toio(volatile void __iomem *to, const void *from, size_t n)
 {
-	if (cc_platform_has(CC_ATTR_GUEST_UNROLL_STRING_IO))
+	if (IS_ENABLED(CONFIG_KOBOX_HOSTED) || cc_platform_has(CC_ATTR_GUEST_UNROLL_STRING_IO))
 		unrolled_memcpy_toio(to, from, n);
 	else
 		string_memcpy_toio(to, from, n);
@@ -113,7 +113,7 @@
 
 void memset_io(volatile void __iomem *a, int b, size_t c)
 {
-	if (cc_platform_has(CC_ATTR_GUEST_UNROLL_STRING_IO)) {
+	if (IS_ENABLED(CONFIG_KOBOX_HOSTED) || cc_platform_has(CC_ATTR_GUEST_UNROLL_STRING_IO)) {
 		unrolled_memset_io(a, b, c);
 	} else {
 		/*
